float64 distance
---
bool result
